src/Pages/SolvingQuestion.tsx

```diff
@@ -2,138 +2,154 @@ import { faArrowLeft } from '@fortawesome/free-solid-svg-icons'
 
 export function SolvingQuestion() {
   const navigate = useNavigate()
-  const [options, setOptions] = useState(sampleOptions)
-  const [selectedOption, setSelectedOption] = useState<number | null>(null)
-  const [isAnswered, setIsAnswered] = useState(false)
-  const [answer, setAnswer] = useState<number | null>(null)
+  const qid = useParams().id
+  const cid = useParams().cid
+  const uid = useSelector((state: RootState) => state.userInfo._id)
+  const [optionSet, setOptionSet] = useState<optionType[]>()
+  const [options, setOptions] = useState([])
+  const [qinfo, setQinfo] = useState<qinfoType>()
+  const [ansVisible, setAnsVisible] = useState(true)
+  const [selected, setSelected] = useState<number>()
+  const [answer, setAnswer] = useState(0)
+  const [isSolved, setIsSolved] = useState(false)
+  const [showAnswer, setShowAnswer] = useState(false)
   const [isOpenModal, setIsOpenModal] = useState<boolean>(false)
 
-  /* CONNECTING DB AFTER FINISHING STATE CONTROL(BUILDING....)
-    const qid = useParams().id;
-	const cid = useParams().cid;
-    /* CONNECTING DB AFTER FINISHING STATE CONTROL(BUILDING....)
-
-    function shuffle(arr:Array<any>) {
-        return [...arr].sort(() => 0.5 - Math.random())
-    }
-
-    function getMultipleRandom(arr: Array<any>, num: number) {
-		const shuffled = shuffle(arr);
-		return shuffled.slice(0, num);
-	}
-
-    function getShuffledOptions() {
-        axios.get(`${process.env.REACT_APP_BACK_END}/question/detail/load?qid=` + qid).then((res) => {
-			axios
-				.get(`${process.env.REACT_APP_BACK_END}/question/load/cluster?qid=` + qid)
-				.then((res2) => {
-					const cluster = res2.data.cluster;
-					const ans = cluster.filter((c:any) => c.representative.is_answer).map((e:any) => e.representative);
-					const dis = cluster.filter((c:any) => !c.representative.is_answer).map((e:any) => e.representative);
-					const ansList = getMultipleRandom(ans, 1);
-					const disList = getMultipleRandom(dis, 3);
-
-                    ansRef.current = ans;
-					setOptions(shuffle(ansList.concat(disList)));
-				})
-				.catch((err) => console.log(err));
-            setQInfo(res.data.qinfo);
-		});
-        setSelectedOption('');
-    }
-    */
+  const getMultipleRandom = useCallback((arr: optionType[], num: number) => {
+    const shuffled = [...arr].sort(() => 0.5 - Math.random())
+    return shuffled.slice(0, num)
+  }, [])
 
-  useEffect(() => {
-    setAnswer(0)
+  const shuffle = useCallback((array: optionType[]) => {
+    array.sort(() => Math.random() - 0.5)
+    return array
   }, [])
 
-  const clickOption = useCallback(
-    (i: number) => () => {
-      if (isAnswered == false) {
-        setSelectedOption(i)
-        console.log('sdf', i)
-      }
-    },
-    [selectedOption, isAnswered]
-  )
+  const getQinfo = useCallback((qid: string | undefined) => {
+    let optionList
+    axios.get(`${process.env.REACT_APP_BACK_END}/question/detail/load?qid=` + qid).then(res => {
```

Comment:
search all axios and please change them to `Post` or `Get`

Reply:
> search all axios and please change them to `Post` or `Get`

Alright :)